### ISSUE TITLE
Connection reset/unable to generate token for user

### ISSUE THREAD

**Issue by schifazl** [PowerShell/Win32-OpenSSH#1106]
Hi!
I have the latest openssh (7.6.0.0) installed on a windows 2008 server and a windows 7 (client). I'm at work, in a domain, and trying to connect to the server without password.

Connecting by using my password works flawlessly, but after generating the private/public key pair, thigs go south.

This is the client side log:

> D:\Users\MyUsername>ssh -vvv MyUsername@100.200.300.400
> OpenSSH_7.6p1, OpenSSL 1.0.2n  7 Dec 2017
> debug1: Reading configuration data /etc/ssh/ssh_config
> debug2: resolving "100.200.300.400" port 22
> debug2: ssh_connect_direct: needpriv 0
> debug1: Connecting to 100.200.300.400 [100.200.300.400] port 22.
> debug1: Connection established.
> debug1: identity file /d/Users/MyUsername/.ssh/id_rsa type 0
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_rsa-cert type -1
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_dsa type -1
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_dsa-cert type -1
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_ecdsa type -1
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_ecdsa-cert type -1
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_ed25519 type -1
> debug1: key_load_public: No such file or directory
> debug1: identity file /d/Users/MyUsername/.ssh/id_ed25519-cert type -1
> debug1: Local version string SSH-2.0-OpenSSH_7.6
> debug1: Remote protocol version 2.0, remote software version OpenSSH_for_Windows_7.6
> debug1: match: OpenSSH_for_Windows_7.6 pat OpenSSH* compat 0x04000000
> debug2: fd 3 setting O_NONBLOCK
> debug1: Authenticating to 100.200.300.400:22 as 'MyUsername'
> debug3: hostkeys_foreach: reading file "/d/Users/MyUsername/.ssh/known_hosts"
> debug3: record_hostkey: found key type ECDSA in file /d/Users/MyUsername/.ssh/known_hosts:1
> debug3: load_hostkeys: loaded 1 keys from 100.200.300.400
> debug3: order_hostkeyalgs: prefer hostkeyalgs: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
> debug3: send packet: type 20
> debug1: SSH2_MSG_KEXINIT sent
> debug3: receive packet: type 20
> debug1: SSH2_MSG_KEXINIT received
> debug2: local client KEXINIT proposal
> debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,ext-info-c
> debug2: host key algorithms: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
> debug2: ciphers ctos: aes128-ctr,aes192-ctr,aes256-ctr,aes128-cbc,3des-cbc,aes256-cbc,aes192-cbc
> debug2: ciphers stoc: aes128-ctr,aes192-ctr,aes256-ctr,aes128-cbc,3des-cbc,aes256-cbc,aes192-cbc
> debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
> debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
> debug2: compression ctos: none,zlib@openssh.com,zlib
> debug2: compression stoc: none,zlib@openssh.com,zlib
> debug2: languages ctos:
> debug2: languages stoc:
> debug2: first_kex_follows 0
> debug2: reserved 0
> debug2: peer server KEXINIT proposal
> debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1
> debug2: host key algorithms: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
> debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr
> debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr
> debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
> debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
> debug2: compression ctos: none
> debug2: compression stoc: none
> debug2: languages ctos:
> debug2: languages stoc:
> debug2: first_kex_follows 0
> debug2: reserved 0
> debug1: kex: algorithm: curve25519-sha256
> debug1: kex: host key algorithm: ecdsa-sha2-nistp256
> debug1: kex: server->client cipher: aes128-ctr MAC: umac-64-etm@openssh.com compression: none
> debug1: kex: client->server cipher: aes128-ctr MAC: umac-64-etm@openssh.com compression: none
> debug3: send packet: type 30
> debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
> debug3: receive packet: type 31
> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:S1NHzF8YXwaSrvkVfg8XeXaYIGoDio6x6+D2kvRpSVQ
> debug3: hostkeys_foreach: reading file "/d/Users/MyUsername/.ssh/known_hosts"
> debug3: record_hostkey: found key type ECDSA in file /d/Users/MyUsername/.ssh/known_hosts:1
> debug3: load_hostkeys: loaded 1 keys from 100.200.300.400
> debug1: Host '100.200.300.400' is known and matches the ECDSA host key.
> debug1: Found key in /d/Users/MyUsername/.ssh/known_hosts:1
> debug3: send packet: type 21
> debug2: set_newkeys: mode 1
> debug1: rekey after 4294967296 blocks
> debug1: SSH2_MSG_NEWKEYS sent
> debug1: expecting SSH2_MSG_NEWKEYS
> debug3: receive packet: type 21
> debug1: SSH2_MSG_NEWKEYS received
> debug2: set_newkeys: mode 0
> debug1: rekey after 4294967296 blocks
> debug2: key: /d/Users/MyUsername/.ssh/id_rsa (0x600063900)
> debug2: key: /d/Users/MyUsername/.ssh/id_dsa (0x0)
> debug2: key: /d/Users/MyUsername/.ssh/id_ecdsa (0x0)
> debug2: key: /d/Users/MyUsername/.ssh/id_ed25519 (0x0)
> debug3: send packet: type 5
> debug3: receive packet: type 7
> debug1: SSH2_MSG_EXT_INFO received
> debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
> debug3: receive packet: type 6
> debug2: service_accept: ssh-userauth
> debug1: SSH2_MSG_SERVICE_ACCEPT received
> debug3: send packet: type 50
> debug3: receive packet: type 51
> debug1: Authentications that can continue: publickey,password,keyboard-interactive
> debug3: start over, passed a different list publickey,password,keyboard-interactive
> debug3: preferred publickey,keyboard-interactive,password
> debug3: authmethod_lookup publickey
> debug3: remaining preferred: keyboard-interactive,password
> debug3: authmethod_is_enabled publickey
> debug1: Next authentication method: publickey
> debug1: Offering public key: RSA SHA256:EY7RzIErmmVLilfO8Be2AZyhTaYWhIXowkdl9pKwWok /d/Users/MyUsername/.ssh/id_rsa
> debug3: send_pubkey_test
> debug3: send packet: type 50
> debug2: we sent a publickey packet, wait for reply
> debug3: receive packet: type 60
> debug1: Server accepts key: pkalg rsa-sha2-512 blen 279
> debug2: input_userauth_pk_ok: fp SHA256:EY7RzIErmmVLilfO8Be2AZyhTaYWhIXowkdl9pKwWok
> debug3: sign_and_send_pubkey: RSA SHA256:EY7RzIErmmVLilfO8Be2AZyhTaYWhIXowkdl9pKwWok
> debug3: send packet: type 50
> debug3: receive packet: type 52
> debug1: Authentication succeeded (publickey).
> Authenticated to 100.200.300.400 ([100.200.300.400]:22).
> debug1: channel 0: new [client-session]
> debug3: ssh_session2_open: channel_new: 0
> debug2: channel 0: send open
> debug3: send packet: type 90
> debug1: Requesting no-more-sessions@openssh.com
> debug3: send packet: type 80
> debug1: Entering interactive session.
> debug1: pledge: network
> debug3: send packet: type 1
> Connection reset by 100.200.300.400 port 22

And this is the sshd.log part:

> 5528 2018-03-15 13:54:20.931 error: unable to generate token for user le00034
> 5528 2018-03-15 13:54:20.947 error: unable to generate token on 2nd attempt for user le00034
> 5528 2018-03-15 13:54:20.947 error: unable to get security token for user le00034
> 5528 2018-03-15 13:54:20.947 error: privsep_postauth, posix_spawn failed

I've already tries, without success:

- Using username@domain@host to login (I get "invalid user" in the sshd.log)
- Adding myself to the "Access this computer from the network" section in the "Local Security Policy" window

Any suggestion? Thanks!

**Comment by manojampalam** [comment]
Can you see if key based auth works for a local user? If it does, can you double check that your domain controller is accessible? User tokens for key based auth are generated via S4U - generating domain user S4U tokens requires a persistent connection with DC.

**Comment by schifazl** [comment]
Thanks for your quick reply!

I've created a local user and it worked flawlessly. I've also checked if the DC is accessible by using
`nltest /dsgetdc:<domain name>` and it also worked.

**Comment by schifazl** [comment]
Hello!
Still no luck with my account.

Any idea?

**Comment by manojampalam** [comment]
Can you use v1.0.0.0 version and try out using FQDN
ssh user@abc.def.com@target 
and share sshd logs if it doesn't work?

Reason to try the prior version - to avoid hitting https://github.com/PowerShell/Win32-OpenSSH/issues/1098

**Comment by schifazl** [comment]
Still no luck.

Strange thing: I tried to edit the sshd_config file by explicitly adding `PubkeyAuthentication yes` and `PasswordAuthentication no`, but it seems that the configuration is ignored, as the public  key method will allways fail and fall back to the password auth.

**Comment by manojampalam** [comment]
Please share logs with v1.0.0.0. Note that the location of sshd_config and logs has changed between v1.0.0.0 and v7.6.0.0

**Comment by rkeithhill** [comment]
I'm running into the same issue w/7.6.0.0 - trying to connect using my domain account on the same machine that sshd is running on.

**Comment by bagajjal** [comment]
Please debug further,
https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Comment by rkeithhill** [comment]
Looks like I'm running into #1098.
```
debug3: NetUserGetInfo() failed with error: 1722 for user: hillr and domain: acme.com

DsGetDcNameW() failed with error: 1355

Invalid user hillr@acme.com from ::1 port 1101
```
Any ETA on an update with that fix?

**Comment by NoMoreFood** [comment]
@rkeithhill I have a pull request in that's being evaluated that might address this issue.  I can compile a custom sshd.exe for testing in my fork if you're interested/willing.

**Comment by rkeithhill** [comment]
In looking at issues what were fixed in today's 7.6.1.0p1-Beta, I had assumed this issue was fixed. See https://github.com/PowerShell/Win32-OpenSSH/issues/1098

**Comment by bagajjal** [comment]
#1098  is fixed in the recent release 7.6.1.0p1. please try the latest release.

**Comment by bagajjal** [comment]
Reopen if the issue is not fixed.
